File: menu.py
from app import get_index_html
from notes import add_note, list_notes


def main_menu() -> str:
    print("Main Menu")
    print("1. Show index page")
    print("2. Notes")
    print("3. Exit")
    return input("Select an option: ")


def notes_menu() -> str:
    print("Notes Menu")
    print("1. Add note")
    print("2. List notes")
    print("3. Back")
    return input("Select an option: ")


def main() -> None:
    while True:
        choice = main_menu()
        if choice == '1':
            print(get_index_html())
        elif choice == '2':
            while True:
                n_choice = notes_menu()
                if n_choice == '1':
                    note = input("Enter note: ").strip()
                    if note:
                        add_note(note)
                elif n_choice == '2':
                    list_notes()
                else:
                    break
        elif choice == '3':
            break
        else:
            print("Invalid option")


if __name__ == '__main__':
    main()

File: menu.php
<?php
require_once __DIR__ . '/notes.php';

function mainMenu(): void {
    echo "Main Menu\n";
    echo "1. Show index page\n";
    echo "2. Notes\n";
    echo "3. Exit\n";
    echo "Select an option: ";
}

function notesMenu(): void {
    echo "Notes Menu\n";
    echo "1. Add note\n";
    echo "2. List notes\n";
    echo "3. Back\n";
    echo "Select an option: ";
}

while (true) {
    mainMenu();
    $choice = trim(fgets(STDIN));

    switch ($choice) {
        case '1':
            // Include index.php to output the HTML
            require __DIR__ . '/index.php';
            break;
        case '2':
            while (true) {
                notesMenu();
                $nChoice = trim(fgets(STDIN));
                if ($nChoice === '1') {
                    echo "Enter note: ";
                    $note = trim(fgets(STDIN));
                    if ($note !== '') {
                        addNote($db, $note);
                    }
                } elseif ($nChoice === '2') {
                    listNotes($db);
                } else {
                    break;
                }
            }
            break;
        case '3':
            exit(0);
        default:
            echo "Invalid option\n";
    }
}
